<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>De novo transcriptome assembly Workflow Template • systemPipeMetaTrans</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="De novo transcriptome assembly Workflow Template">
<meta property="og:description" content="systemPipeMetaTrans">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158796879-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158796879-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeMetaTrans</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/systemPipeMetaTrans.html">
    <span class="fas fa fas fa-book"></span>
     
    Workflow
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/systemPipeMetaTrans">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>De novo transcriptome assembly Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 29 April, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/systemPipeMetaTrans/blob/master/vignettes/systemPipeMetaTrans.Rmd"><code>vignettes/systemPipeMetaTrans.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeMetaTrans.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!-- 
Compile from command-line  
Rscript -e "rmarkdown::render('systemPipeDeNovo.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeDeNovo.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeDeNovo.knit.md 13; Rscript -e "rmarkdown::render('systemPipeDeNovo.Rmd', c('BiocStyle::pdf_document'))"
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Users want to provide here background information about the CLIP-Seq project.</p>
<div id="background-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#background-and-objectives" class="anchor"></a>Background and objectives</h2>
<p>This report describes the analysis of several e-CLIP experiments studying the RNA biding proteins.</p>
</div>
<div id="experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-design" class="anchor"></a>Experimental design</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div id="workflow-environment" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-environment" class="anchor"></a>Workflow environment</h1>
<div id="environment-settings-and-input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#environment-settings-and-input-data" class="anchor"></a>Environment settings and input data</h2>
<p>Typically, the user wants to record here the sources and versions of the reference genome sequence along with the corresponding annotations. In the provided sample data set all data inputs are stored in a <code>data</code> subdirectory and all results will be written to a separate <code>results</code> directory, while the <code>systemPipeRNAseq.Rmd</code> script and the <code>targets</code> file are expected to be located in the parent directory. The R session is expected to run from this parent directory.</p>
<p>To run this sample report, mini sample FASTQ and reference genome files can be downloaded from <a href="http://biocluster.ucr.edu/~tgirke/projects/systemPipeR_test_data.zip">here</a>. The chosen data set <a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRP010938">SRP010938</a> contains 18 paired-end (PE) read sets from <em>Arabidposis thaliana</em> <span class="citation">(Howard et al. 2013)</span>. To minimize processing time during testing, each FASTQ file has been subsetted to 90,000-100,000 randomly sampled PE reads that map to the first 100,000 nucleotides of each chromosome of the <em>A. thalina</em> genome. The corresponding reference genome sequence (FASTA) and its GFF annotation files (provided in the same download) have been truncated accordingly. This way the entire test sample data set is less than 200MB in storage space. A PE read set has been chosen for this test data set for flexibility, because it can be used for testing both types of analysis routines requiring either SE (single end) reads or PE reads.</p>
<p>Instructions for loading the demo data available for NGS workflow templates into the user’s current working directory can be found here. At the moment, the package includes workflow templates for RNA-Seq, ChIP-Seq, VAR-Seq and Ribo-Seq. Templates for additional NGS applications will be provided in the future.</p>
<p>Alternatively, this can be done from the command-line as find here.</p>
<p>Now open the R markdown script <code>systemPipeRNAseq.Rmd</code>in your R IDE (<em>e.g.</em> vim-r or RStudio) and run the workflow as outlined below. If you work under Vim-R-Tmux, the following command sequence will connect the user in an interactive session with a node on the cluster [TODO: fix, point to the main vignette]. The code of the <code>Rmd</code>script can then be sent from Vim on the login (head) node to an open R session running on the corresponding computer node. This is important since Tmux sessions should not be run on the computer nodes.</p>
<p>Now check whether your R session is running on a computer node of the cluster and not on a head node.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"hostname"</span>) <span class="co"># should return name of a compute node starting with i or c </span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>() <span class="co"># checks current working directory of R session</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>() <span class="co"># returns content of current working directory</span></pre></body></html></div>
</div>
<div id="required-packages-and-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#required-packages-and-resources" class="anchor"></a>Required packages and resources</h2>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">systemPipeR</span>)</pre></body></html></div>
<p>If applicable load custom functions not provided by <code>systemPipeR</code> package.</p>
<p>To apply workflows to custom data, the user needs to modify the <em><code>targets</code></em> file and if necessary update the corresponding <em><code>.cwl</code></em> and <em><code>.yml</code></em> files. A collection of pre-generated <em><code>.cwl</code></em> and <em><code>.yml</code></em> files are provided in the <em><code>param/cwl</code></em> subdirectory of each workflow template. They are also viewable in the GitHub repository of <em><code>systemPipeRdata</code></em> (<a href="https://github.com/tgirke/systemPipeRdata/tree/master/inst/extdata/param">see here</a>).</p>
</div>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">targetspath</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"systemPipeR"</span>)
<span class="no">targets</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="no">targetspath</span>, <span class="kw">comment.char</span> <span class="kw">=</span> <span class="st">"#"</span>)[,<span class="fl">1</span>:<span class="fl">4</span>]
<span class="no">targets</span></pre></body></html></div>
<pre><code>##                       FileName SampleName Factor SampleLong
## 1  ./data/SRR446027_1.fastq.gz        M1A     M1  Mock.1h.A
## 2  ./data/SRR446028_1.fastq.gz        M1B     M1  Mock.1h.B
## 3  ./data/SRR446029_1.fastq.gz        A1A     A1   Avr.1h.A
## 4  ./data/SRR446030_1.fastq.gz        A1B     A1   Avr.1h.B
## 5  ./data/SRR446031_1.fastq.gz        V1A     V1   Vir.1h.A
## 6  ./data/SRR446032_1.fastq.gz        V1B     V1   Vir.1h.B
## 7  ./data/SRR446033_1.fastq.gz        M6A     M6  Mock.6h.A
## 8  ./data/SRR446034_1.fastq.gz        M6B     M6  Mock.6h.B
## 9  ./data/SRR446035_1.fastq.gz        A6A     A6   Avr.6h.A
## 10 ./data/SRR446036_1.fastq.gz        A6B     A6   Avr.6h.B
## 11 ./data/SRR446037_1.fastq.gz        V6A     V6   Vir.6h.A
## 12 ./data/SRR446038_1.fastq.gz        V6B     V6   Vir.6h.B
## 13 ./data/SRR446039_1.fastq.gz       M12A    M12 Mock.12h.A
## 14 ./data/SRR446040_1.fastq.gz       M12B    M12 Mock.12h.B
## 15 ./data/SRR446041_1.fastq.gz       A12A    A12  Avr.12h.A
## 16 ./data/SRR446042_1.fastq.gz       A12B    A12  Avr.12h.B
## 17 ./data/SRR446043_1.fastq.gz       V12A    V12  Vir.12h.A
## 18 ./data/SRR446044_1.fastq.gz       V12B    V12  Vir.12h.B</code></pre>
</div>
</div>
<div id="read-preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#read-preprocessing" class="anchor"></a>Read preprocessing</h1>
<div id="read-quality-filtering-and-trimming" class="section level2">
<h2 class="hasAnchor">
<a href="#read-quality-filtering-and-trimming" class="anchor"></a>Read quality filtering and trimming</h2>
<p>The function <code>preprocessReads</code> allows to apply predefined or custom read preprocessing functions to all FASTQ files referenced in a <code>SYSargs2</code> container, such as quality filtering or adapter trimming routines. The paths to the resulting output FASTQ files are stored in the <code>output</code> slot of the <code>SYSargs2</code> object. The following example performs adapter trimming with the <code>trimLRPatterns</code> function from the <code>Biostrings</code> package. After the trimming step a new targets file is generated (here <code>targets_trim.txt</code>) containing the paths to the trimmed FASTQ files. The new targets file can be used for the next workflow step with an updated <code>SYSargs2</code> instance, <em>e.g.</em> running the NGS alignments using the trimmed FASTQ files.</p>
<p>Construct <em><code>SYSargs2</code></em> object from <em><code>cwl</code></em> and <em><code>yml</code></em> param and <em><code>targets</code></em> files.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">dir_path</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/preprocessReads/trim-se"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"systemPipeR"</span>)
<span class="no">trim</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="kw">targets</span><span class="kw">=</span><span class="no">targetspath</span>, <span class="kw">wf_file</span><span class="kw">=</span><span class="st">"trim-se.cwl"</span>, <span class="kw">input_file</span><span class="kw">=</span><span class="st">"trim-se.yml"</span>, <span class="kw">dir_path</span><span class="kw">=</span><span class="no">dir_path</span>)
<span class="no">trim</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(<span class="no">trim</span>, <span class="kw">inputvars</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">FileName</span><span class="kw">=</span><span class="st">"_FASTQ_PATH1_"</span>, <span class="kw">SampleName</span><span class="kw">=</span><span class="st">"_SampleName_"</span>))
<span class="no">trim</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span>(<span class="no">trim</span>)[<span class="fl">1</span>:<span class="fl">2</span>]</pre></body></html></div>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/preprocessReads.html">preprocessReads</a></span>(<span class="kw">args</span><span class="kw">=</span><span class="no">trim</span>, <span class="kw">Fct</span><span class="kw">=</span><span class="st">"trimLRPatterns(Rpattern='GCCCGGGTAA', 
                subject=fq)"</span>, <span class="kw">batchsize</span><span class="kw">=</span><span class="fl">100000</span>, <span class="kw">overwrite</span><span class="kw">=</span><span class="fl">TRUE</span>, <span class="kw">compress</span><span class="kw">=</span><span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">trim</span>, <span class="kw">file</span><span class="kw">=</span><span class="st">"targets_trimPE.txt"</span>, <span class="kw">step</span> <span class="kw">=</span> <span class="fl">1</span>,
                <span class="kw">new_col</span> <span class="kw">=</span> <span class="st">"FileName1"</span>, <span class="kw">new_col_output_index</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">overwrite</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
</div>
<div id="fastq-quality-report" class="section level2">
<h2 class="hasAnchor">
<a href="#fastq-quality-report" class="anchor"></a>FASTQ quality report</h2>
<p>The following <code>seeFastq</code> and <code>seeFastqPlot</code> functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box plots, base proportions, base-level quality trends, relative k-mer diversity, length and occurrence distribution of reads, number of reads above quality cutoffs and mean quality distribution. The results are written to a PDF file named <code>fastqReport.pdf</code>.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">fqlist</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastq</a></span>(<span class="kw">fastq</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(<span class="no">trim</span>), <span class="kw">batchsize</span><span class="kw">=</span><span class="fl">10000</span>, <span class="kw">klength</span><span class="kw">=</span><span class="fl">8</span>)
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/fastqReport.pdf"</span>, <span class="kw">height</span><span class="kw">=</span><span class="fl">18</span>, <span class="kw">width</span><span class="kw">=</span><span class="fl">4</span>*<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">fqlist</span>))
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastqPlot</a></span>(<span class="no">fqlist</span>)
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</pre></body></html></div>
<img src="results/fastqReport.png"><div align="center">
Figure 1: FASTQ quality report for 18 samples
</div>
<p></p>
</div>
</div>
<div id="metarna-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#metarna-workflow" class="anchor"></a>MetaRNA workflow</h1>
<p><strong>GOAL</strong> Metatranscriptome analysis is challenging and many different tools and workflows have been developed, but every procedure has its pros and cons. Here we introduce # MetaRNA workflow</p>
<p><strong>GOAL</strong> Metatranscriptome analysis is challenging and many different tools and workflows have been developed, but every procedure has its pros and cons. Here we introduce an interactive workflow that allows the user to build its own automated workflow within the package systemPipeR. Currently, the options available for the user are:</p>
<ul>
<li>Soft or regular quality trimming</li>
<li>Filter reads by taxon (i.e. remove unwanted organisms) or use all data</li>
<li>
<em>de novo</em> assembly or genome-guided assembly</li>
<li>Read count normalization on global scale or re-scaled from organism-level normalization</li>
<li>Annotation</li>
<li>Differential expression gene or transcript analysis</li>
</ul>
<div id="kaiju" class="section level2">
<h2 class="hasAnchor">
<a href="#kaiju" class="anchor"></a>Kaiju</h2>
</div>
<div id="de-novo-assembly-with-trinity" class="section level2">
<h2 class="hasAnchor">
<a href="#de-novo-assembly-with-trinity" class="anchor"></a><em>De novo</em> assembly with Trinity</h2>
</div>
<div id="quantification-with-salmon" class="section level2">
<h2 class="hasAnchor">
<a href="#quantification-with-salmon" class="anchor"></a>Quantification with Salmon</h2>
</div>
<div id="summarizing-counts-and-calculating-exn50-stats" class="section level2">
<h2 class="hasAnchor">
<a href="#summarizing-counts-and-calculating-exn50-stats" class="anchor"></a>Summarizing counts and calculating ExN50 stats</h2>
</div>
<div id="annotation-with-trinotate" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-with-trinotate" class="anchor"></a>Annotation with Trinotate</h2>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R Under development (unstable) (2020-03-31 r78116)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04 LTS
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so
## LAPACK: /usr/local/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] batchtools_0.9.13           ape_5.3                     ggplot2_3.3.0              
##  [4] systemPipeR_1.21.9          ShortRead_1.45.4            GenomicAlignments_1.23.2   
##  [7] SummarizedExperiment_1.17.5 DelayedArray_0.13.12        matrixStats_0.56.0         
## [10] Biobase_2.47.3              BiocParallel_1.21.3         Rsamtools_2.3.7            
## [13] Biostrings_2.55.7           XVector_0.27.2              GenomicRanges_1.39.3       
## [16] GenomeInfoDb_1.23.17        IRanges_2.21.8              S4Vectors_0.25.15          
## [19] BiocGenerics_0.33.3         BiocStyle_2.15.8           
## 
## loaded via a namespace (and not attached):
##  [1] colorspace_1.4-1         rjson_0.2.20             hwriter_1.3.2           
##  [4] ellipsis_0.3.0           rprojroot_1.3-2          fs_1.4.1                
##  [7] bit64_0.9-7              AnnotationDbi_1.49.1     codetools_0.2-16        
## [10] splines_4.1.0            knitr_1.28               annotate_1.65.1         
## [13] GO.db_3.10.0             dbplyr_1.4.3             png_0.1-7               
## [16] pheatmap_1.0.12          graph_1.65.3             BiocManager_1.30.10     
## [19] compiler_4.1.0           httr_1.4.1               GOstats_2.53.0          
## [22] backports_1.1.6          assertthat_0.2.1         Matrix_1.2-18           
## [25] limma_3.43.8             htmltools_0.4.0          prettyunits_1.1.1       
## [28] tools_4.1.0              gtable_0.3.0             glue_1.4.0              
## [31] GenomeInfoDbData_1.2.3   Category_2.53.1          dplyr_0.8.5             
## [34] rappdirs_0.3.1           Rcpp_1.0.4.6             pkgdown_1.5.1           
## [37] vctrs_0.2.4              nlme_3.1-145             rtracklayer_1.47.0      
## [40] xfun_0.13                stringr_1.4.0            lifecycle_0.2.0         
## [43] XML_3.99-0.3             edgeR_3.29.1             zlibbioc_1.33.1         
## [46] MASS_7.3-51.5            scales_1.1.0             BSgenome_1.55.4         
## [49] VariantAnnotation_1.33.4 hms_0.5.3                RBGL_1.63.1             
## [52] RColorBrewer_1.1-2       yaml_2.2.1               curl_4.3                
## [55] memoise_1.1.0            biomaRt_2.43.6           latticeExtra_0.6-29     
## [58] stringi_1.4.6            RSQLite_2.2.0            genefilter_1.69.0       
## [61] desc_1.2.0               checkmate_2.0.0          GenomicFeatures_1.39.7  
## [64] rlang_0.4.5              pkgconfig_2.0.3          bitops_1.0-6            
## [67] evaluate_0.14            lattice_0.20-40          purrr_0.3.4             
## [70] bit_1.1-15.2             tidyselect_1.0.0         GSEABase_1.49.1         
## [73] AnnotationForge_1.29.3   magrittr_1.5             bookdown_0.18           
## [76] R6_2.4.1                 base64url_1.4            DBI_1.1.0               
## [79] pillar_1.4.3             withr_2.2.0              survival_3.1-11         
## [82] RCurl_1.98-1.2           tibble_3.0.1             crayon_1.3.4            
## [85] BiocFileCache_1.11.6     rmarkdown_2.1            jpeg_0.1-8.1            
## [88] progress_1.2.2           locfit_1.5-9.4           grid_4.1.0              
## [91] data.table_1.12.8        blob_1.2.1               Rgraphviz_2.31.0        
## [94] digest_0.6.25            xtable_1.8-4             brew_1.0-6              
## [97] openssl_1.4.1            munsell_0.5.0            askpass_1.1</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-H_Backman2016-bt">
<p>H Backman, Tyler W, and Thomas Girke. 2016. “systemPipeR: NGS workflow and report generation environment.” <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.</p>
</div>
<div id="ref-Howard2013-fq">
<p>Howard, Brian E, Qiwen Hu, Ahmet Can Babaoglu, Manan Chandra, Monica Borghi, Xiaoping Tan, Luyan He, et al. 2013. “High-Throughput RNA Sequencing of Pseudomonas-Infected Arabidopsis Reveals Hidden Transcriptome Complexity and Novel Splice Variants.” <em>PLoS One</em> 8 (10): e74183. <a href="https://doi.org/10.1371/journal.pone.0074183">https://doi.org/10.1371/journal.pone.0074183</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '',
    indexName: '',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
